finality-aleph/src/data_io/data_provider.rs

```diff
@@ -148,7 +148,15 @@ where
         // of `data_to_propose` unaffected.
 
         let client_info = self.client.info();
-        let finalized_block = (client_info.finalized_hash, client_info.finalized_number).into();
+        let finalized_block: BlockHashNum<B> =
+            (client_info.finalized_hash, client_info.finalized_number).into();
+
+        if finalized_block.num >= self.session_boundaries.last_block() {
+            // This session is already finished, but this instance of ChainTracker has not been terminated yet.
+            // We go with the default -- empty proposal, this does not have any significance.
+            *self.data_to_propose.lock().await = AlephData::Empty;
```

Comment:
i do not want to be seen as mental, but it seems that a small testcase would be kinda appropriate - just to expose this case, which occurred to be easily overlooked

Reply:
Added a test case.